src/components/cc-token-oauth-list/cc-token-oauth-list.js

```diff
@@ -0,0 +1,368 @@
+ */
+export class CcTokenOauthList extends LitElement {
+  static get properties() {
+    return {
+      state: { type: Object },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {TokenOauthListState} The state of the component */
+    this.state = { type: 'loading' };
+
+    /** @type {RefHTMLDivElement} */
+    this._emptyMessageRef = createRef();
+
+    new ResizeController(this, {
+      widthBreakpoints: [995, 730],
+    });
+
+    new LostFocusController(
+      this,
+      '.oauth-token-card__action-revoke, .revoke-all-tokens-button',
+      ({ suggestedElement }) => {
+        if (suggestedElement instanceof HTMLElement) {
+          suggestedElement.focus();
+        } else {
+          this._emptyMessageRef.value.focus();
+        }
+      },
+    );
+  }
+
+  _onRevokeAllOauthTokens() {
+    dispatchCustomEvent(this, 'revoke-all-oauth-tokens');
+  }
+
+  /** @param {string} tokenId */
+  _onRevokeToken(tokenId) {
+    dispatchCustomEvent(this, 'revoke-oauth-token', tokenId);
+  }
+
+  render() {
+    if (this.state.type === 'error') {
+      return html`<cc-notice intent="warning" message="${i18n('cc-token-oauth-list.error')}"></cc-notice>`;
+    }
+
+    const hasOauthTokens =
+      (this.state.type === 'loaded' || this.state.type === 'revoking-all') && this.state.oauthTokens.length > 0;
+    const isEmpty = this.state.type === 'loaded' && this.state.oauthTokens.length === 0;
+    const sortedOauthTokens =
+      this.state.type === 'loaded' || this.state.type === 'revoking-all'
+        ? [...this.state.oauthTokens].sort(
+            (tokenA, tokenB) => tokenB.creationDate.getTime() - tokenA.creationDate.getTime(),
+          )
+        : [];
+
+    return html`
+      <cc-block>
+        <div slot="header-title">${i18n('cc-token-oauth-list.main-heading')}</div>
+        <div slot="header-right">
+          ${hasOauthTokens
+            ? html`
+                <cc-button
+                  class="revoke-all-tokens-button"
+                  danger
+                  outlined
+                  ?waiting=${this.state.type === 'revoking-all'}
+                  @cc-button:click=${this._onRevokeAllOauthTokens}
+                >
+                  ${i18n('cc-token-oauth-list.revoke-all-tokens')}
+                </cc-button>
+              `
+            : ''}
+        </div>
+        <div slot="content">
+          <p>${i18n('cc-token-oauth-list.intro')}</p>
+          <div class="oauth-token-list-wrapper">
+            ${this.state.type === 'loading' ? html`<cc-loader></cc-loader>` : ''}
+            ${isEmpty
+              ? html`<p class="empty" tabindex="-1" ${ref(this._emptyMessageRef)}>
+                  ${i18n('cc-token-oauth-list.empty')}
+                </p>`
+              : ''}
+            <div class="oauth-token-list-wrapper__list">
+              ${this.state.type === 'loaded' || this.state.type === 'revoking-all'
```

Comment:
nitpick: maybe you don't need this condition since `sortedOauthTokens` is already constrained.